clang/lib/CConv/ConstraintResolver.cpp

```diff
@@ -358,10 +358,25 @@ std::set<ConstraintVariable *>
         /* Allocator call */
         if (isFunctionAllocator(FD->getName())) {
           bool didInsert = false;
-          // FIXME: Should be treating malloc, realloc, calloc differently
+          // FIXME: Add support for realloc
           if (CE->getNumArgs() > 0) {
             QualType ArgTy;
-            ConstAtom *A = analyzeAllocExpr(CE->getArg(0), CS, ArgTy);
+            std::string FuncName = FD->getNameAsString();
+            ConstAtom *A;
+            if (!FuncName.compare("calloc")) {
```

Comment:
Can we move this code (for handling`calloc`) into the `analyzeAllocExpr` function? That's what `calloc` is, after all. :) You could just put this at the start of the function and then continue on with what's there.